Error in librenms.log during automatic polling.

### The problem

Hello,

While polling devices using a cron job, I found this message in the LibreNMS log. I haven't noticed any malfunction in the system itself, but this message concerns me. It looks like a bug where the polling process tries to write something to the database that doesn't belong in the table. Could you please advise on what to do about this? There is more than what I'm sending, but this is for illustration.

Thank you for your help.
```
[2025-03-03T13:15:51][ALERT] INFO: device:poll MAR03V.io.test (143) polled in 3.696s
[2025-03-03T13:15:52][ERROR] SQLSTATE[22007]: Invalid datetime format: 1366 Incorrect double value: 'INF' for column `librenms`.`application_metrics`.`value` at row 1 (Connection: mysql, SQL: UPDATE `application_metrics` set `value`=INF,`value_prev`=3 WHERE app_id=28 && metric=tracking_reference_name) (Connection: dbFacile, SQL: UPDATE `application_metrics` set `value`=INF,`value_prev`=3 WHERE app_id=28 && metric=tracking_reference_name)#0 /usr/local/www/librenms.io.test/includes/polling/functions.inc.php(355): dbUpdate()
#1 /usr/local/www/librenms.io.test/includes/polling/applications/chronyd.inc.php(111): update_application()
#2 /usr/local/www/librenms.io.test/includes/polling/applications.inc.php(11): include('...')
#3 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Collections/Traits/EnumeratesValues.php(240): LibreNMS\Modules\LegacyModule->{closure:/usr/local/www/librenms.io.test/includes/polling/applications.inc.php:6}()
#4 /usr/local/www/librenms.io.test/includes/polling/applications.inc.php(6): Illuminate\Support\Collection->each()
#5 /usr/local/www/librenms.io.test/LibreNMS/Modules/LegacyModule.php(114): include('...')
#6 /usr/local/www/librenms.io.test/app/Jobs/PollDevice.php(140): LibreNMS\Modules\LegacyModule->poll()
#7 /usr/local/www/librenms.io.test/app/Jobs/PollDevice.php(64): App\Jobs\PollDevice->pollModules()
#8 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36): App\Jobs\PollDevice->handle()
#9 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Container/Util.php(41): Illuminate\Container\BoundMethod::{closure:Illuminate\Container\BoundMethod::call():35}()
#10 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(93): Illuminate\Container\Util::unwrapIfClosure()
#11 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(35): Illuminate\Container\BoundMethod::callBoundMethod()
#12 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Container/Container.php(662): Illuminate\Container\BoundMethod::call()
#13 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php(128): Illuminate\Container\Container->call()
#14 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(144): Illuminate\Bus\Dispatcher->{closure:Illuminate\Bus\Dispatcher::dispatchNow():125}()
#15 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(119): Illuminate\Pipeline\Pipeline->{closure:Illuminate\Pipeline\Pipeline::prepareDestination():142}()
#16 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php(132): Illuminate\Pipeline\Pipeline->then()
#17 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Queue/CallQueuedHandler.php(123): Illuminate\Bus\Dispatcher->dispatchNow()
#18 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(144): Illuminate\Queue\CallQueuedHandler->{closure:Illuminate\Queue\CallQueuedHandler::dispatchThroughMiddleware():122}()
#19 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(119): Illuminate\Pipeline\Pipeline->{closure:Illuminate\Pipeline\Pipeline::prepareDestination():142}()
#20 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Queue/CallQueuedHandler.php(122): Illuminate\Pipeline\Pipeline->then()
#21 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Queue/CallQueuedHandler.php(70): Illuminate\Queue\CallQueuedHandler->dispatchThroughMiddleware()
#22 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Queue/Jobs/Job.php(102): Illuminate\Queue\CallQueuedHandler->call()
#23 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Queue/SyncQueue.php(43): Illuminate\Queue\Jobs\Job->fire()
#24 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php(254): Illuminate\Queue\SyncQueue->push()
#25 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php(230): Illuminate\Bus\Dispatcher->pushCommandToQueue()
#26 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php(95): Illuminate\Bus\Dispatcher->dispatchToQueue()
#27 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Foundation/Bus/Dispatchable.php(76): Illuminate\Bus\Dispatcher->dispatchSync()
#28 /usr/local/www/librenms.io.test/app/Console/Commands/DevicePoll.php(79): App\Jobs\PollDevice::dispatchSync()
#29 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36): App\Console\Commands\DevicePoll->handle()
#30 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Container/Util.php(41): Illuminate\Container\BoundMethod::{closure:Illuminate\Container\BoundMethod::call():35}()
#31 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(93): Illuminate\Container\Util::unwrapIfClosure()
#32 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(35): Illuminate\Container\BoundMethod::callBoundMethod()
#33 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Container/Container.php(662): Illuminate\Container\BoundMethod::call()
#34 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Console/Command.php(211): Illuminate\Container\Container->call()
#35 /usr/local/www/librenms.io.test/vendor/symfony/console/Command/Command.php(326): Illuminate\Console\Command->execute()
#36 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Console/Command.php(180): Symfony\Component\Console\Command\Command->run()
#37 /usr/local/www/librenms.io.test/vendor/symfony/console/Application.php(1096): Illuminate\Console\Command->run()
#38 /usr/local/www/librenms.io.test/vendor/symfony/console/Application.php(324): Symfony\Component\Console\Application->doRunCommand()
#39 /usr/local/www/librenms.io.test/vendor/symfony/console/Application.php(175): Symfony\Component\Console\Application->doRun()
#40 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(201): Symfony\Component\Console\Application->run()
#41 /usr/local/www/librenms.io.test/app/Console/Kernel.php(66): Illuminate\Foundation\Console\Kernel->handle()
#42 /usr/local/www/librenms.io.test/lnms(38): App\Console\Kernel->handle()
#43 {main}
```

```
[2025-03-03T13:15:42][ALERT] INFO: device:poll JNNS1.io.test (123) polled in 4.037s
[2025-03-03T13:15:45][ERROR] SQLSTATE[22007]: Invalid datetime format: 1366 Incorrect double value: 'INF' for column `librenms`.`application_metrics`.`value` at row 1 (Connection: mysql, SQL: UPDATE `application_metrics` set `value`=INF,`value_prev`=3 WHERE app_id=34 && metric=tracking_reference_name) (Connection: dbFacile, SQL: UPDATE `application_metrics` set `value`=INF,`value_prev`=3 WHERE app_id=34 && metric=tracking_reference_name)#0 /usr/local/www/librenms.io.test/includes/polling/functions.inc.php(355): dbUpdate()
#1 /usr/local/www/librenms.io.test/includes/polling/applications/chronyd.inc.php(111): update_application()
#2 /usr/local/www/librenms.io.test/includes/polling/applications.inc.php(11): include('...')
#3 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Collections/Traits/EnumeratesValues.php(240): LibreNMS\Modules\LegacyModule->{closure:/usr/local/www/librenms.io.test/includes/polling/applications.inc.php:6}()
#4 /usr/local/www/librenms.io.test/includes/polling/applications.inc.php(6): Illuminate\Support\Collection->each()
#5 /usr/local/www/librenms.io.test/LibreNMS/Modules/LegacyModule.php(114): include('...')
#6 /usr/local/www/librenms.io.test/app/Jobs/PollDevice.php(140): LibreNMS\Modules\LegacyModule->poll()
#7 /usr/local/www/librenms.io.test/app/Jobs/PollDevice.php(64): App\Jobs\PollDevice->pollModules()
#8 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36): App\Jobs\PollDevice->handle()
#9 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Container/Util.php(41): Illuminate\Container\BoundMethod::{closure:Illuminate\Container\BoundMethod::call():35}()
#10 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(93): Illuminate\Container\Util::unwrapIfClosure()
#11 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(35): Illuminate\Container\BoundMethod::callBoundMethod()
#12 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Container/Container.php(662): Illuminate\Container\BoundMethod::call()
#13 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php(128): Illuminate\Container\Container->call()
#14 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(144): Illuminate\Bus\Dispatcher->{closure:Illuminate\Bus\Dispatcher::dispatchNow():125}()
#15 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(119): Illuminate\Pipeline\Pipeline->{closure:Illuminate\Pipeline\Pipeline::prepareDestination():142}()
#16 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php(132): Illuminate\Pipeline\Pipeline->then()
#17 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Queue/CallQueuedHandler.php(123): Illuminate\Bus\Dispatcher->dispatchNow()
#18 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(144): Illuminate\Queue\CallQueuedHandler->{closure:Illuminate\Queue\CallQueuedHandler::dispatchThroughMiddleware():122}()
#19 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(119): Illuminate\Pipeline\Pipeline->{closure:Illuminate\Pipeline\Pipeline::prepareDestination():142}()
#20 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Queue/CallQueuedHandler.php(122): Illuminate\Pipeline\Pipeline->then()
#21 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Queue/CallQueuedHandler.php(70): Illuminate\Queue\CallQueuedHandler->dispatchThroughMiddleware()
#22 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Queue/Jobs/Job.php(102): Illuminate\Queue\CallQueuedHandler->call()
#23 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Queue/SyncQueue.php(43): Illuminate\Queue\Jobs\Job->fire()
#24 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php(254): Illuminate\Queue\SyncQueue->push()
#25 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php(230): Illuminate\Bus\Dispatcher->pushCommandToQueue()
#26 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php(95): Illuminate\Bus\Dispatcher->dispatchToQueue()
#27 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Foundation/Bus/Dispatchable.php(76): Illuminate\Bus\Dispatcher->dispatchSync()
#28 /usr/local/www/librenms.io.test/app/Console/Commands/DevicePoll.php(79): App\Jobs\PollDevice::dispatchSync()
#29 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36): App\Console\Commands\DevicePoll->handle()
#30 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Container/Util.php(41): Illuminate\Container\BoundMethod::{closure:Illuminate\Container\BoundMethod::call():35}()
#31 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(93): Illuminate\Container\Util::unwrapIfClosure()
#32 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(35): Illuminate\Container\BoundMethod::callBoundMethod()
#33 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Container/Container.php(662): Illuminate\Container\BoundMethod::call()
#34 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Console/Command.php(211): Illuminate\Container\Container->call()
#35 /usr/local/www/librenms.io.test/vendor/symfony/console/Command/Command.php(326): Illuminate\Console\Command->execute()
#36 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Console/Command.php(180): Symfony\Component\Console\Command\Command->run()
#37 /usr/local/www/librenms.io.test/vendor/symfony/console/Application.php(1096): Illuminate\Console\Command->run()
#38 /usr/local/www/librenms.io.test/vendor/symfony/console/Application.php(324): Symfony\Component\Console\Application->doRunCommand()
#39 /usr/local/www/librenms.io.test/vendor/symfony/console/Application.php(175): Symfony\Component\Console\Application->doRun()
#40 /usr/local/www/librenms.io.test/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(201): Symfony\Component\Console\Application->run()
#41 /usr/local/www/librenms.io.test/app/Console/Kernel.php(66): Illuminate\Foundation\Console\Kernel->handle()
#42 /usr/local/www/librenms.io.test/lnms(38): App\Console\Kernel->handle()
#43 {main}
```


### Output of ./validate.php

```txt
===========================================
Component | Version
--------- | -------
LibreNMS  | 25.2.0-30-g6e2b4e384 (2025-03-03T11:17:32+01:00)
DB Schema | 2025_01_30_214311_create_ipv6_nd_table (320)
PHP       | 8.4.4
Python    | 3.9.21
Database  | MariaDB 10.5.27-MariaDB
RRDTool   | 1.7.2
SNMP      | 5.9.1
===========================================

[OK]    Composer Version: 2.8.6
[OK]    Dependencies up-to-date.
[OK]    Database connection successful
[OK]    Database connection successful
[OK]    Database Schema is current
[OK]    SQL Server meets minimum requirements
[OK]    lower_case_table_names is enabled
[OK]    MySQL engine is optimal
[OK]    Database and column collations are correct
[OK]    Database schema correct
[OK]    MySQL and PHP time match
[OK]    Distributed Polling setting is enabled globally
[OK]    Connected to rrdcached
[OK]    Active pollers found
[OK]    Dispatcher Service not detected
[OK]    Locks are functional
[OK]    Python poller wrapper is polling
[OK]    Redis is unavailable
[OK]    rrdtool version ok
[OK]    Connected to rrdcached
```

### What was the last working version of LibreNMS?

25.2.0-30-g6e2b4e384 (2025-03-03T11:17:32+01:00)

### Anything in the logs that might be useful for us?

```txt

```

--- a/includes/polling/applications/chronyd.inc.php
+++ b/includes/polling/applications/chronyd.inc.php
@@ -65,7 +65,6 @@ $source_rrd_def = RrdDefinition::make()
     ->addDataset('offset', 'GAUGE', -1000, 1000)
     ->addDataset('stddev', 'GAUGE', -1000, 1000);
 
-
 foreach ($chronyd['sources'] as $source) {
     $sources[] = $source['source_name'];
     $rrd_name = ['app', $name, $app->app_id, $source['source_name']];

--- a/includes/polling/applications/chronyd.inc.php
+++ b/includes/polling/applications/chronyd.inc.php
@@ -42,6 +42,8 @@ $fields = [
     'update_interval' => $chronyd['tracking']['update_interval'],
 ];
 
+$metrics = $fields;
+
 $tags = ['name' => $name, 'app_id' => $app->app_id, 'rrd_def' => $rrd_def, 'rrd_name' => $rrd_name];
 app('Datastore')->put($device, 'app', $tags, $fields);
 
@@ -63,8 +65,6 @@ $source_rrd_def = RrdDefinition::make()
     ->addDataset('offset', 'GAUGE', -1000, 1000)
     ->addDataset('stddev', 'GAUGE', -1000, 1000);
 
-$metrics = $chronyd;
-unset($metrics['sources']);
 
 foreach ($chronyd['sources'] as $source) {
     $sources[] = $source['source_name'];
@@ -87,11 +87,6 @@ foreach ($chronyd['sources'] as $source) {
 
     $tags = ['name' => $name, 'app_id' => $app->app_id, 'rrd_def' => $source_rrd_def, 'rrd_name' => $rrd_name];
     app('Datastore')->put($device, 'app', $tags, $fields);
-
-    // insert flattened source metrics into the metrics array
-    foreach ($fields as $field => $value) {
-        $metrics['source_' . $source['source_name'] . '_' . $field] = $value;
-    }
 }
 
 // check for added or removed sources

--- a/tests/data/linux_chronyd.json
+++ b/tests/data/linux_chronyd.json
@@ -29,85 +29,67 @@
             ],
             "application_metrics": [
                 {
-                    "metric": "tracking_frequency",
+                    "metric": "frequency",
                     "value": -9.845,
                     "value_prev": null,
                     "app_type": "chronyd"
                 },
                 {
-                    "metric": "tracking_last_offset",
+                    "metric": "last_offset",
                     "value": 1.5e-8,
                     "value_prev": null,
                     "app_type": "chronyd"
                 },
                 {
-                    "metric": "tracking_leap_status",
-                    "value": 0,
-                    "value_prev": null,
-                    "app_type": "chronyd"
-                },
-                {
-                    "metric": "tracking_reference_name",
-                    "value": 50505300,
-                    "value_prev": null,
-                    "app_type": "chronyd"
-                },
-                {
-                    "metric": "tracking_reference_time",
+                    "metric": "reference_time",
                     "value": 1613135374.7654,
                     "value_prev": null,
                     "app_type": "chronyd"
                 },
                 {
-                    "metric": "tracking_reference_type",
+                    "metric": "residual_frequency",
                     "value": 0,
                     "value_prev": null,
                     "app_type": "chronyd"
                 },
                 {
-                    "metric": "tracking_residual_frequency",
-                    "value": 0,
-                    "value_prev": null,
-                    "app_type": "chronyd"
-                },
-                {
-                    "metric": "tracking_rms_offset",
+                    "metric": "rms_offset",
                     "value": 9.5e-8,
                     "value_prev": null,
                     "app_type": "chronyd"
                 },
                 {
-                    "metric": "tracking_root_delay",
+                    "metric": "root_delay",
                     "value": 1.0e-9,
                     "value_prev": null,
                     "app_type": "chronyd"
                 },
                 {
-                    "metric": "tracking_root_dispersion",
+                    "metric": "root_dispersion",
                     "value": 8.054e-6,
                     "value_prev": null,
                     "app_type": "chronyd"
                 },
                 {
-                    "metric": "tracking_skew",
+                    "metric": "skew",
                     "value": 0.006,
                     "value_prev": null,
                     "app_type": "chronyd"
                 },
                 {
-                    "metric": "tracking_stratum",
+                    "metric": "stratum",
                     "value": 1,
                     "value_prev": null,
                     "app_type": "chronyd"
                 },
                 {
-                    "metric": "tracking_system_time",
+                    "metric": "system_time",
                     "value": -1.6e-8,
                     "value_prev": null,
                     "app_type": "chronyd"
                 },
                 {
-                    "metric": "tracking_update_interval",
+                    "metric": "update_interval",
                     "value": 8,
                     "value_prev": null,
                     "app_type": "chronyd"

